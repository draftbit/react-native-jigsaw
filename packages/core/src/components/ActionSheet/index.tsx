export { default as ActionSheet } from "./ActionSheet";
export { default as ActionSheetItem } from "./ActionSheetItem";
export { default as ActionSheetCancel } from "./ActionSheetCancel";
