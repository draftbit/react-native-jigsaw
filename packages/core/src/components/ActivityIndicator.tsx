import * as React from "react";
import {
  StyleProp,
  ViewStyle,
  ActivityIndicator as ActivityIndicatorRN,
  View,
  Platform,
} from "react-native";
import { withTheme } from "@draftbit/theme";
import type { ReadTheme } from "@draftbit/theme";
import {
  Bounce,
  Chase,
  Circle,
  CircleFade,
  Flow,
  Fold,
  Grid,
  Plane,
  Pulse,
  Swing,
  Wander,
  Wave,
} from "react-native-animated-spinkit";

export enum ActivityIndicatorType {
  default = "default",
  plane = "plane",
  chase = "chase",
  bounce = "bounce",
  wave = "wave",
  pulse = "pulse",
  flow = "flow",
  swing = "swing",
  circle = "circle",
  circleFade = "circleFade",
  grid = "grid",
  fold = "fold",
  wander = "wander",
}

type Props = {
  style?: StyleProp<ViewStyle>;
  color?: string;
  theme: ReadTheme;
  type?: ActivityIndicatorType;
  size?: number | "small" | "large";
};

const SPINNER_COMPONENTS = {
  [ActivityIndicatorType.plane]: Plane,
  [ActivityIndicatorType.chase]: Chase,
  [ActivityIndicatorType.bounce]: Bounce,
  [ActivityIndicatorType.wave]: Wave,
  [ActivityIndicatorType.pulse]: Pulse,
  [ActivityIndicatorType.flow]: Flow,
  [ActivityIndicatorType.swing]: Swing,
  [ActivityIndicatorType.circle]: Circle,
  [ActivityIndicatorType.circleFade]: CircleFade,
  [ActivityIndicatorType.grid]: Grid,
  [ActivityIndicatorType.fold]: Fold,
  [ActivityIndicatorType.wander]: Wander,
};

const smallActivityIndicatorSize = 20;
const largeActivityIndicatorSize = 40;

const getLoadingSizeNumber = (size?: number | "small" | "large"): number => {
  if (typeof size === "number") return size;
  return size === "large"
    ? largeActivityIndicatorSize
    : smallActivityIndicatorSize;
};

const getScaleLevel = (size?: number | "small" | "large"): number => {
  if (typeof size === "number") return size / smallActivityIndicatorSize;
  return size === "large" ? 2 : 1;
};

const ActivityIndicator: React.FC<React.PropsWithChildren<Props>> = ({
  theme,
  color = theme.colors.branding.primary,
  type = ActivityIndicatorType.default,
  size = "small",
  style,
  ...rest
}) => {
  const sizeNumber = getLoadingSizeNumber(size);
  const spinnerColor = color ?? theme.colors.branding.primary;

  // Handle display spinner type
  if (type !== ActivityIndicatorType.default) {
    if (type && SPINNER_COMPONENTS[type]) {
      const SpinnerComponent = SPINNER_COMPONENTS[type];
      return (
        <SpinnerComponent
          size={sizeNumber}
          color={spinnerColor}
          style={style}
        />
      );
    }
  }

  // This is due to limitations with iOS UIActivityIndicator
  // Refer: https://github.com/facebook/react-native/issues/12250
  const scaleLevel = getScaleLevel(size);
  if (Platform.OS === "ios") {
    return (
      <View
        style={[
          style,
          {
            transform: [{ scale: scaleLevel }],
          },
        ]}
      >
        <ActivityIndicatorRN
          animating={true}
          hidesWhenStopped={true}
          size={"small"}
          color={spinnerColor}
          {...rest}
        />
      </View>
    );
  }
  return (
    <ActivityIndicatorRN
      animating={true}
      hidesWhenStopped={true}
      size={size}
      color={spinnerColor}
      style={style}
      {...rest}
    />
  );
};

export default withTheme(ActivityIndicator);
