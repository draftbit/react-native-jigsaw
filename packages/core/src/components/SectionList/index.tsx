export { default as SectionList } from "./SectionList";
export { default as SectionHeader } from "./SectionHeader";
export type {
  FlatListSectionListProps,
  FlashListSectionListProps,
} from "./SectionList";
