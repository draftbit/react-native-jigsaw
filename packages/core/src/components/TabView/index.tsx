export { default as TabView } from "./TabView";
export { default as TabViewItem } from "./TabViewItem";
