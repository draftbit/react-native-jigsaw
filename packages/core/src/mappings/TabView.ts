import {
  COMPONENT_TYPES,
  createTextEnumProp,
  createColorProp,
  StylesPanelSections,
  Triggers,
  createActionProp,
  createStaticBoolProp,
  GROUPS,
} from "@draftbit/types";

export const SEED_DATA = {
  name: "Tab View",
  tag: "TabView",
  description: "Tab view container",
  category: COMPONENT_TYPES.swiper,
  stylesPanelSections: [
    StylesPanelSections.Size,
    StylesPanelSections.MarginsAndPaddings,
    StylesPanelSections.Position,
    StylesPanelSections.Effects,
    StylesPanelSections.LayoutSelectedItem,
    StylesPanelSections.Background,
    StylesPanelSections.Typography,
  ],
  triggers: [Triggers.OnIndexChanged, Triggers.OnEndReached],
  props: {
    onIndexChanged: createActionProp({
      label: "On index changed",
      description: "Action to execute when swipe to new index",
    }),
    onEndReached: createActionProp({
      label: "On end reached",
      description: "Action to execute when end of swiping reached",
    }),
    tabBarPosition: createTextEnumProp({
      label: "Tab Bar Position",
      description: "Position of tab bar",
      options: ["top", "bottom"],
      defaultValue: "top",
    }),
    keyboardDismissMode: createTextEnumProp({
      label: "Keyboard Dismiss Mode",
      description: "Keyboard Dismiss Mode",
      options: ["auto", "on-drag", "none"],
      defaultValue: "auto",
    }),
    swipeEnabled: createStaticBoolProp({
      label: "Swipe Enabled",
      description: "Whether swiping is enabled or not",
      defaultValue: true,
    }),
    scrollEnabled: createStaticBoolProp({
      label: "Tab Scroll Enabled",
      description: "Whether scrolling of tabs is enabled or not",
      defaultValue: false,
    }),
    activeColor: createColorProp({
      label: "Active Color",
      description: "Color of icon and text of active tab",
      defaultValue: "primary",
    }),
    inactiveColor: createColorProp({
      label: "Inactive Color",
      description: "Color of icon and text of inactive tab(s)",
      defaultValue: null,
    }),
    pressColor: createColorProp({
      label: "Press Color",
      description: "Color of ripple when pressed",
      defaultValue: "primary",
      group: GROUPS.android,
    }),
    indicatorColor: createColorProp({
      label: "Indicator Color",
      description: "Color of indicator",
      defaultValue: "primary",
    }),
    tabsBackgroundColor: createColorProp({
      label: "Tabs Background Color",
      description: "Background color of tabs container",
      defaultValue: "background",
    }),
  },
};
