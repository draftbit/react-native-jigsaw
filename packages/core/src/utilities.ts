import { StyleSheet, StyleProp, TextStyle } from "react-native";
import { isString, isNumber, pick, pickBy, identity } from "lodash";

export function extractStyles(style: StyleProp<any>) {
  const {
    color,
    fontFamily,
    fontWeight,
    fontSize,
    lineHeight,
    letterSpacing,
    textTransform,
    textAlign,
    textDecorationLine,
    textDecorationColor,
    textDecorationStyle,
    ...viewStyles
  } = StyleSheet.flatten(style || {});

  const textStyles: TextStyle = {
    color,
    fontFamily,
    fontWeight,
    fontSize,
    lineHeight,
    letterSpacing,
    textTransform,
    textAlign,
    textDecorationLine,
    textDecorationColor,
    textDecorationStyle,
  };

  return { viewStyles, textStyles };
}

export const borderStyleNames = [
  "borderRadius",
  "borderBottomColor",
  "borderBottomEndRadius",
  "borderBottomLeftRadius",
  "borderBottomRightRadius",
  "borderBottomStartRadius",
  "borderBottomWidth",
  "borderColor",
  "borderEndColor",
  "borderLeftColor",
  "borderLeftWidth",
  "borderRadius",
  "borderRightColor",
  "borderRightWidth",
  "borderStartColor",
  "borderStyle",
  "borderTopColor",
  "borderTopEndRadius",
  "borderTopLeftRadius",
  "borderTopRightRadius",
  "borderTopStartRadius",
  "borderTopWidth",
  "borderWidth",
];

export const marginStyleNames = [
  "margin",
  "marginBottom",
  "marginEnd",
  "marginHorizontal",
  "marginLeft",
  "marginRight",
  "marginStart",
  "marginTop",
  "marginVertical",
];

export function extractBorderAndMarginStyles(
  style: StyleProp<any>,
  additionalBorderStyles?: string[],
  additionalMarginStyles?: string[]
) {
  const flatStyle = StyleSheet.flatten(style || {});

  const borderStyles = pickBy(
    pick(flatStyle, [
      ...borderStyleNames,
      ...(additionalBorderStyles ? additionalBorderStyles : []),
    ]),
    identity
  );

  const marginStyles = pickBy(
    pick(flatStyle, [
      ...marginStyleNames,
      ...(additionalMarginStyles ? additionalMarginStyles : []),
    ]),
    identity
  );

  return { borderStyles, marginStyles };
}

/**
 * Merges a style object on top of another style object. In React Native,
 * keys with undefined values in a style object will still override styles
 * that appear earlier in a sequence. This avoids that problem.
 *
 * This lets us avoid the `...(something ? { something } : {})` pattern.
 * There doesn't seem to be a better way to do this. These all seem to not
 * work (i.e. they all result in `{ color: undefined }`:
 * `const mergedStyles = [{ color: "red" }, { color: undefined }]`
 * `const mergedStyles = StyleSheet.compose({ color: "red" }, { color: undefined })`
 * `const mergedStyles = StyleSheet.flatten([{ color: "red" }, { color: undefined }])`
 */
export function applyStyles(
  baseStyles: Array<StyleProp<any>>,
  stylesToApply: StyleProp<any> | undefined
) {
  if (!stylesToApply) {
    return;
  }

  const flattenedStyles = StyleSheet.flatten(baseStyles);

  for (const [key, value] of Object.entries(stylesToApply)) {
    if (value != null) {
      flattenedStyles[key] = value;
    }
  }

  return flattenedStyles;
}

export function getValueForRadioButton(value: string | number) {
  if (isString(value)) {
    return value;
  } else if (isNumber(value)) {
    return String(value);
  } else {
    throw new Error(`Invalid value: ${value}`);
  }
}
