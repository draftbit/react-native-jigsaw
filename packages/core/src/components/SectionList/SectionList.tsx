import React from "react";
import { FlashListProps, FlashList } from "@shopify/flash-list";
import { FlatListProps, FlatList } from "react-native";
import SectionHeader, { DefaultSectionHeader } from "./SectionHeader";
import { flattenReactFragments } from "../../utilities";

type ListComponentType = "FlatList" | "FlashList";

interface AdditionalSectionListProps<T> {
  sectionKey: string;
  stickyHeader?: boolean;
  renderItem: (itemInfo: {
    item?: T;
    index: number;
    section: string;
  }) => JSX.Element;
  keyExtractor?: (item: T, index: number) => string;
  listComponent?: ListComponentType;
}

export type FlatListSectionListProps<T> = Omit<FlatListProps<T>, "renderItem"> &
  AdditionalSectionListProps<T>;

export type FlashListSectionListProps<T> = Omit<
  FlashListProps<T>,
  "renderItem"
> &
  AdditionalSectionListProps<T>;

interface SectionListDataItem<T> {
  type: "DATA_ITEM";
  data: T;
}

interface SectionListSectionItem {
  type: "SECTION_ITEM";
  title: string;
}

type SectionListItem<T> = SectionListDataItem<T> | SectionListSectionItem;

export const DEFAULT_SECTION = "Uncategorized";

const SectionList = <T extends { [key: string]: any }>({
  sectionKey,
  stickyHeader = false,
  listComponent = "FlatList",
  data: dataProp,
  renderItem: renderItemProp,
  keyExtractor: keyExtractorProp,
  ...rest
}: FlatListSectionListProps<T> | FlashListSectionListProps<T>) => {
  const data = React.useMemo(
    () => (Array.isArray(dataProp) ? dataProp : []) as T[],
    [dataProp]
  );

  const dataWithSections = React.useMemo(() => {
    const result: SectionListItem<T>[] = [];
    const sectionDataItems: { [key: string]: T[] } = {};

    for (const item of data) {
      const section = item[sectionKey]?.toString() || DEFAULT_SECTION;
      if (sectionDataItems[section]) {
        sectionDataItems[section].push(item);
      } else {
        sectionDataItems[section] = [item];
      }
    }

    for (const section in sectionDataItems) {
      result.push({ type: "SECTION_ITEM", title: section });
      const sectionItems: SectionListDataItem<T>[] = sectionDataItems[
        section
      ].map((item) => ({ type: "DATA_ITEM", data: item }));
      result.push(...sectionItems);
    }

    return result;
  }, [data, sectionKey]);

  const sectionHeaderIndicies = React.useMemo(
    () =>
      stickyHeader
        ? dataWithSections
            .filter((item) => item.type === "SECTION_ITEM")
            .map((item) => dataWithSections.indexOf(item))
        : undefined,
    [dataWithSections, stickyHeader]
  );

  const extractSectionHeader = (
    element: JSX.Element | null
  ): JSX.Element | null => {
    if (!element) {
      return null;
    }

    const props = element.props || {};
    const children = flattenReactFragments(
      React.Children.toArray(props.children) as React.ReactElement[]
    );

    if (element.type === SectionHeader) {
      return element;
    } else {
      for (const child of children) {
        if (child.type === SectionHeader) {
          return child;
        }
      }
    }
    return null;
  };

  const extractRemainingNonSectionHeader = (
    element: JSX.Element | null
  ): JSX.Element | null => {
    if (!element) {
      return null;
    }

    const props = element.props || {};
    const children = flattenReactFragments(
      React.Children.toArray(props.children) as React.ReactElement[]
    );
    if (element.type === SectionHeader) {
      return null;
    } else {
      const newChildren = [];
      for (const child of children) {
        if (child.type !== SectionHeader) {
          newChildren.push(child);
        }
      }
      return React.cloneElement(element, {
        ...props,
        children: newChildren,
      });
    }
  };

  const renderItem = ({
    item,
    index,
  }: {
    item: SectionListItem<T>;
    index: number;
  }) => {
    switch (item.type) {
      case "SECTION_ITEM": {
        const renderedItem = renderItemProp({
          index,
          section: item.title,
        });
        return (
          extractSectionHeader(renderedItem) || (
            <DefaultSectionHeader title={item.title} />
          )
        );
      }
      case "DATA_ITEM": {
        const renderedItem = renderItemProp({
          item: item.data,
          index,
          section: item.data[sectionKey] || DEFAULT_SECTION,
        });
        return extractRemainingNonSectionHeader(renderedItem);
      }
    }
  };

  const keyExtractor = (item: SectionListItem<T>, index: number) => {
    switch (item.type) {
      case "SECTION_ITEM": {
        return `section_${index.toString()}`;
      }
      case "DATA_ITEM": {
        return keyExtractorProp?.(item.data, index) || index.toString();
      }
    }
  };

  switch (listComponent) {
    case "FlatList":
      return (
        <FlatList
          stickyHeaderIndices={sectionHeaderIndicies}
          {...(rest as FlatListProps<SectionListItem<T>>)}
          data={dataWithSections}
          renderItem={renderItem}
          keyExtractor={keyExtractor}
        />
      );
    case "FlashList":
      return (
        <FlashList
          stickyHeaderIndices={sectionHeaderIndicies}
          {...(rest as FlashListProps<SectionListItem<T>>)}
          data={dataWithSections}
          renderItem={renderItem}
          keyExtractor={keyExtractor}
        />
      );
  }
};

export default SectionList;
