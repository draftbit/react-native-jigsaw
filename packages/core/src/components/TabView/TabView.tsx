import * as React from "react";
import { StyleProp, ViewStyle } from "react-native";
import {
  TabView,
  TabBar,
  SceneRendererProps,
  NavigationState,
  Route,
} from "react-native-tab-view";

import TabViewItem from "./TabViewItem";
import type { IconSlot } from "../../interfaces/Icon";
import { withTheme } from "../../theming";
import type { Theme } from "../../styles/DefaultTheme";
import { extractIfNestedInFragment, extractStyles } from "../../utilities";

type SceneProps = SceneRendererProps & {
  route: Route;
};
type TabBarProps = SceneRendererProps & {
  navigationState: NavigationState<any>;
};
type TabBarPosition = "top" | "bottom";
type KeyboardDismissMode = "none" | "auto" | "on-drag";

type TabViewProps = {
  onIndexChanged?: (index: number) => void;
  onEndReached?: () => void;
  initialTabIndex?: number;
  tabBarPosition?: TabBarPosition;
  keyboardDismissMode?: KeyboardDismissMode;
  swipeEnabled?: boolean;
  scrollEnabled?: boolean;
  activeColor?: string;
  inactiveColor?: string;
  pressColor?: string;
  indicatorColor?: string;
  tabsBackgroundColor?: string;
  style?: StyleProp<ViewStyle>;
  theme: Theme;
} & IconSlot;

const TabViewComponent: React.FC<React.PropsWithChildren<TabViewProps>> = ({
  Icon,
  onIndexChanged,
  onEndReached,
  initialTabIndex = 0,
  tabBarPosition,
  keyboardDismissMode,
  swipeEnabled,
  scrollEnabled,
  activeColor,
  inactiveColor,
  pressColor,
  indicatorColor,
  tabsBackgroundColor,
  style,
  theme,
  children: childrenProp,
}) => {
  const [index, setIndex] = React.useState(initialTabIndex);
  const [routes, setRoutes] = React.useState<Route[]>([]);
  const [tabScenes, setTabScenes] = React.useState<{ [key: string]: any }>({});

  const { textStyles, viewStyles } = extractStyles(style);

  const children: React.ReactNode[] = React.useMemo(
    () =>
      React.Children.toArray(childrenProp).map((child) =>
        extractIfNestedInFragment(child as React.ReactElement)
      ),
    [childrenProp]
  );

  //Populate routes and scenes based on children
  React.useEffect(() => {
    const newRoutes: Route[] = [];
    const scenes: { [key: string]: React.ReactElement } = {};

    children
      .filter(
        (child) => React.isValidElement(child) && child.type === TabViewItem
      )
      .forEach((item: any, idx) => {
        const child = item as React.ReactElement;
        newRoutes.push({
          key: idx.toString(),
          title: child.props.title,
          icon: child.props.icon,
          accessibilityLabel: child.props.accessibilityLabel,
        });
        scenes[idx] = child;
      });

    setRoutes(newRoutes);
    setTabScenes(scenes);
  }, [children]);

  const indexChangeHandler = (i: number) => {
    setIndex(i);
    onIndexChanged?.(i);
    if (i === routes.length) {
      onEndReached?.();
    }
  };

  const renderTabBar: React.FC<TabBarProps> = (props) => {
    return (
      <TabBar
        {...props}
        activeColor={activeColor || theme.colors.primary}
        inactiveColor={inactiveColor || "rgb(200,200,200)"}
        pressColor={pressColor || theme.colors.primary}
        scrollEnabled={scrollEnabled}
        indicatorStyle={{
          backgroundColor: indicatorColor || theme.colors.primary,
        }}
        labelStyle={textStyles}
        renderIcon={({ route, color }) =>
          route?.icon ? (
            <Icon name={route.icon} color={color} size={16} />
          ) : null
        }
        style={{
          backgroundColor: tabsBackgroundColor || theme.colors.background,
        }}
      />
    );
  };

  const renderScene = ({ route }: SceneProps) => {
    return tabScenes[route.key];
  };

  //Cannot render TabView without at least one tab
  if (!routes.length) {
    return <></>;
  }

  return (
    <TabView
      style={viewStyles}
      navigationState={{ index, routes }}
      renderScene={renderScene}
      renderTabBar={renderTabBar}
      onIndexChange={indexChangeHandler}
      tabBarPosition={tabBarPosition}
      keyboardDismissMode={keyboardDismissMode}
      swipeEnabled={swipeEnabled}
    />
  );
};

export default withTheme(TabViewComponent);
