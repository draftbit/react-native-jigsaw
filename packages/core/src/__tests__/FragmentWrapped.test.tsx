import React from "react";
import { View } from "react-native";
import { render, screen } from "@testing-library/react-native";
import { SectionList, SectionHeader } from "../components/SectionList";
import { TabView, TabViewItem } from "../components/TabView";
import {
  SwipeableItem,
  SwipeableItemButton,
} from "../components/SwipeableItem";
import { flattenReactFragments } from "../utilities";

describe("Type checked components wrapped in a fragment tests", () => {
  describe("Type checked components render when wrapped in a fragment tests", () => {
    test("should SectionHeader render when wrapped in fragment", () => {
      render(
        <SectionList
          data={[{ test: "data" }]}
          renderItem={() => (
            <>
              <SectionHeader>
                <View testID="header" />
              </SectionHeader>
            </>
          )}
          sectionKey="test"
        />
      );

      const headerView = screen.queryByTestId("header");
      expect(headerView).toBeTruthy();
    });

    test("should TabViewItem render when wrapped in fragment", () => {
      render(
        <TabView Icon={() => <View />}>
          <>
            <TabViewItem title="Test">
              <View testID="tab-item" />
            </TabViewItem>
          </>
        </TabView>
      );

      const tabItem = screen.queryByTestId("tab-item");
      expect(tabItem).toBeTruthy();
    });

    test("should SwipeableItemButton render when wrapped in fragment", () => {
      render(
        <SwipeableItem Icon={() => <View />}>
          <>
            <SwipeableItemButton revealSwipeDirection="left" title="test" />
          </>
        </SwipeableItem>
      );

      const swipeableButton = screen.queryByTestId("swipeable-behind-item");
      expect(swipeableButton).toBeTruthy();
    });
  });

  describe("flattenReactFragments tests", () => {
    test("should extract components from react fragments", () => {
      const components = [
        <React.Fragment>
          <View testID="1" />
          <View testID="2" />
          <View testID="3" />
        </React.Fragment>,
        <>
          <View testID="4" />
        </>,
        <>
          <View testID="5" />
          <View testID="6" />
        </>,
      ];

      const result = flattenReactFragments(components);
      expect(result).toMatchInlineSnapshot(`
        [
          <View
            testID="1"
          />,
          <View
            testID="2"
          />,
          <View
            testID="3"
          />,
          <View
            testID="4"
          />,
          <View
            testID="5"
          />,
          <View
            testID="6"
          />,
        ]
      `);
    });
  });
});
