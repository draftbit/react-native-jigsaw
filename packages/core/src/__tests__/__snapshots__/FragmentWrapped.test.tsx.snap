// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Type checked components wrapped in a fragment tests flattenReactFragments tests should extract components from react fragments 1`] = `
[
  {
    "$$typeof": Symbol(react.transitional.element),
    "_owner": null,
    "_store": {},
    "key": ".0",
    "props": {
      "testID": "1",
    },
    "type": [Function],
  },
  {
    "$$typeof": Symbol(react.transitional.element),
    "_owner": null,
    "_store": {},
    "key": ".1",
    "props": {
      "testID": "2",
    },
    "type": [Function],
  },
  {
    "$$typeof": Symbol(react.transitional.element),
    "_owner": null,
    "_store": {},
    "key": ".2",
    "props": {
      "testID": "3",
    },
    "type": [Function],
  },
  {
    "$$typeof": Symbol(react.transitional.element),
    "_owner": null,
    "_store": {},
    "key": ".0",
    "props": {
      "testID": "4",
    },
    "type": [Function],
  },
  {
    "$$typeof": Symbol(react.transitional.element),
    "_owner": null,
    "_store": {},
    "key": ".0",
    "props": {
      "testID": "5",
    },
    "type": [Function],
  },
  {
    "$$typeof": Symbol(react.transitional.element),
    "_owner": null,
    "_store": {},
    "key": ".1",
    "props": {
      "testID": "6",
    },
    "type": [Function],
  },
  {
    "$$typeof": Symbol(react.transitional.element),
    "_owner": null,
    "_store": {},
    "key": ".0",
    "props": {
      "testID": "7",
    },
    "type": [Function],
  },
  {
    "$$typeof": Symbol(react.transitional.element),
    "_owner": null,
    "_store": {},
    "key": ".0",
    "props": {
      "testID": "8",
    },
    "type": [Function],
  },
  {
    "$$typeof": Symbol(react.transitional.element),
    "_owner": null,
    "_store": {},
    "key": ".1",
    "props": {
      "testID": "9",
    },
    "type": [Function],
  },
]
`;
